core/src/lib.rs

```diff
@@ -178,7 +183,20 @@ impl Raiko {
         &self,
         origin_pool_txs: Vec<reth_primitives::TransactionSigned>,
         input: &GuestInput,
+        should_execute: bool,
     ) -> RaikoResult<Block> {
+        if should_execute {
+            self.execute_transaction_batch(origin_pool_txs, input)?;
+        }
+
+        Ok(input.block.clone())
+    }
```

Comment:
I have a thought, there are several methods in `Raiko` struct that accepts `should_execute` parameter. This prarameter defined as `let should_execute = proof_type != raiko_lib::proof_type::ProofType::Tdx;`. 

So maybe we could encapsulate this logic inside `Raiko` struct, because it has `raiko.request.proof_type`? 

Meaning we could delete `should_execute` parameter from functions and only change logic inside those functions where it's needed? I'm not sure, so what do you think?

E.g.:
```suggestion
        &self,
        origin_pool_txs: Vec<reth_primitives::TransactionSigned>,
        input: &GuestInput,
    ) -> RaikoResult<Block> {
        let should_execute = self.request.proof_type != raiko_lib::proof_type::ProofType::Tdx;
        if should_execute {
            self.execute_transaction_batch(origin_pool_txs, input)?;
        }

        Ok(input.block.clone())
    }
```